lib/puppet/util/servicenow.rb

```diff
@@ -192,10 +192,49 @@ def report_description(settings_hash, resource_statuses)
     # best and most stable solution we can do (for now) is the description you see here.
     description = "See the PE console for the full report. You can access the PE console at #{settings_hash['pe_console_url']}."
     description << "\n\nResource Statuses:\n#{resourse_status_summary}" unless resourse_status_summary.empty?
+    description << "\n\n== Facts ==\n#{selected_facts(settings_hash)}"
     description
   end
   module_function :report_description
 
+  def facts
+    # This is a cheat to make it easier to do test mocks.
+    Puppet::Node::Facts.indirection.find(host).values
+  end
+
+  def selected_facts(settings_hash, facts_query = nil, format = nil)
+    include_facts = facts_query.nil? ? settings_hash['include_facts'] : facts_query
+    output_format = format.nil? ? settings_hash['facts_format'].to_sym : format.to_sym
+
+    selected_facts = {}
+
+    if [include_facts].flatten.first == 'all'
+      facts
+    else
+      include_facts.each do |name|
+        value = facts.dig(*name.split('.'))
+        selected_facts[name] = value unless value.nil?
+      end
+    end
+
+    # json and object are primarily to support either internal uses like further
+    # processing on selected facts, or giving to the facts to a machine
+    # processor like an event engine.
```

Comment:
This bit seems like it satisfies most of the requirements for PIE-334 as well. Nice!